RAG_supporters/dataset/cluster_labeled_dataset.py

```diff
@@ -0,0 +1,231 @@
+from typing import Any, Dict, Optional, Tuple, Union
+
+import numpy as np
+import torch
+from torch.utils.data import Dataset
+
+from RAG_supporters.dataset.sqlite_storage import SQLiteStorageManager
+
+
+class ClusterLabeledDataset(Dataset):
+    """
+    PyTorch Dataset for domain assessment with 3-type labels.
+    
+    Returns triplets: (base_embedding, steering_embedding, label)
+    
+    Label types:
+    - source_label: Label for base embedding (source/question)
+    - steering_label: Label for steering embedding
+    - combined_label: Weighted average for augmentation masking
+    
+    Storage: SQLite (metadata) + numpy memmap (embeddings).
+    """
+    
+    def __init__(
+        self,
+        dataset_dir: Union[str, Path],
+        label_type: str = "combined",
+        return_metadata: bool = False,
+        mmap_mode: str = 'r'
+    ):
+        """
+        Initialize cluster-labeled dataset.
+        
+        Args:
+            dataset_dir: Directory containing dataset.db and embedding files
+            label_type: Which label to return ('source', 'steering', 'combined')
+            return_metadata: Whether to return metadata dict
+            mmap_mode: Mode for numpy memmap ('r', 'r+', 'w+', 'c')
+        """
+        self.dataset_dir = Path(dataset_dir)
+        self.label_type = label_type
+        self.return_metadata = return_metadata
+        self.mmap_mode = mmap_mode
+        
+        if label_type not in ['source', 'steering', 'combined']:
+            raise ValueError(
+                f"label_type must be 'source', 'steering', or 'combined', got '{label_type}'"
+            )
+        
+        # Open storage
+        db_path = self.dataset_dir / "dataset.db"
+        if not db_path.exists():
+            raise FileNotFoundError(f"Dataset database not found: {db_path}")
+        
+        self.storage = SQLiteStorageManager(db_path)
+        
+        # Load embedding files
+        self.base_embeddings = self._load_embeddings('base')
+        self.steering_embeddings = self._load_embeddings('steering')
+        
+        # Cache samples
+        self.samples = self.storage.get_all_samples()
+        
+        # Load dataset metadata
+        self.n_clusters = self.storage.get_dataset_info('n_clusters')
+        
+        logging.info(f"Loaded ClusterLabeledDataset from {dataset_dir}")
+        logging.info(f"  Samples: {len(self.samples)}")
+        logging.info(f"  Clusters: {self.n_clusters}")
+        logging.info(f"  Label type: {label_type}")
+        logging.info(f"  Base embeddings shape: {self.base_embeddings.shape}")
+        logging.info(f"  Steering embeddings shape: {self.steering_embeddings.shape}")
+    
+    def __len__(self) -> int:
+        """Return dataset size."""
+        return len(self.samples)
+    
+    def __getitem__(self, idx: int) -> Union[Tuple[torch.Tensor, torch.Tensor, torch.Tensor], Tuple[torch.Tensor, torch.Tensor, torch.Tensor, Dict[str, Any]]]:
+        """
```

Comment:
The __getitem__ method has no try-except blocks. If an embedding_idx is invalid or file is corrupted, you get cryptic numpy/torch errors instead of helpful messages.